examples/dwarf_pubnames_types.py

```diff
@@ -0,0 +1,111 @@
+#-------------------------------------------------------------------------------
+# elftools example: dwarf_pubnames_types.py
+#
+# Dump the contents of .debug_pubnames and .debug_pubtypes sections from the
+# ELF file.
+#
+# Note: sample_exe64.elf doesn't have a .debug_pubtypes section.
+#
+# Vijay Ramasami (rvijayc@gmail.com)
+# This code is in the public domain
+#-------------------------------------------------------------------------------
+from __future__ import print_function
+import sys
+
+# If pyelftools is not installed, the example can also run from the root or
+# examples/ dir of the source distribution.
+sys.path[0:0] = ['.', '..']
+
+from elftools.elf.elffile import ELFFile
+
+def process_file(filename):
+    print('Processing file:', filename)
+    with open(filename, 'rb') as f:
+        elffile = ELFFile(f)
+
+        if not elffile.has_dwarf_info():
+            print('  file has no DWARF info')
+            return
+
+        # get_dwarf_info returns a DWARFInfo context object, which is the
+        # starting point for all DWARF-based processing in pyelftools.
+        dwarfinfo = elffile.get_dwarf_info()
+
+        # get .debug_pubtypes section.
+        pubnames = dwarfinfo.get_pubnames()
+        if pubnames is None:
+            print('ERROR: No .debug_pubnames section found in ELF.')
+        else:
+            print('%d entries found in .debug_pubnames' % len(pubnames))
+            
+            # try getting information on a global symbol.
+            print('Trying pubnames example ...')
+            sym_name = 'main'
+            try:
+                entry = pubnames[sym_name]
+            except KeyError:
+                print('ERROR: No pubname entry found for ' + sym_name)
+            else:
+                print('%s: cu_ofs = %d, die_ofs = %d' %
+                        (sym_name, entry.cu_ofs, entry.die_ofs))
+
+                # get the actual CU/DIE that has this information.
+                print('Fetching the actual die for %s ...' % sym_name)
+                for cu in dwarfinfo.iter_CUs():
+                    if cu.cu_offset == entry.cu_ofs:
+                        for die in cu.iter_DIEs():
+                            if die.offset == entry.die_ofs:
+                                print(die)
+        
+            # dump all entries in .debug_pubnames section.
+            print('Dumping .debug_pubnames table ...')
+            print('-'*66)
```

Comment:
Here and everywhere else: spaces around operators 

```
'-' * 66
```

Reply:
I have resolved this in the latest commit.